I suspect that this PR is one of the reasons for the current batch in Tide not compiling:
https://prow.k8s.io/?repo=kubernetes%2Fkubernetes&type=batch&pull=master%2C81168%2C81435%2C81601%2C81624%2C81626
https://prow.k8s.io/log?job=pull-kubernetes-bazel-build&id=1167009731907162126
```
"/b/f/w/pkg/controller/service/service_controller.go:45:2: imported and not used: "k8s.io/kubernetes/pkg/controller"
```

Looks like a bad interaction with some of the other incoming PRs in that batch.

Let me put this PR here on hold for now. Hopefully that'll allow Tide to continue.

/hold

--- a/pkg/controller/service/BUILD
+++ b/pkg/controller/service/BUILD
@@ -16,7 +16,6 @@ go_library(
     importpath = "k8s.io/kubernetes/pkg/controller/service",
     deps = [
         "//pkg/apis/core/v1/helper:go_default_library",
-        "//pkg/controller:go_default_library",
         "//pkg/util/metrics:go_default_library",
         "//staging/src/k8s.io/api/core/v1:go_default_library",
         "//staging/src/k8s.io/apimachinery/pkg/api/errors:go_default_library",

--- a/pkg/controller/service/service_controller.go
+++ b/pkg/controller/service/service_controller.go
@@ -42,7 +42,6 @@ import (
 	servicehelper "k8s.io/cloud-provider/service/helpers"
 	"k8s.io/klog"
 	v1helper "k8s.io/kubernetes/pkg/apis/core/v1/helper"
-	"k8s.io/kubernetes/pkg/controller"
 	"k8s.io/kubernetes/pkg/util/metrics"
 )
 
@@ -195,7 +194,7 @@ func New(
 
 // obj could be an *v1.Service, or a DeletionFinalStateUnknown marker item.
 func (s *ServiceController) enqueueService(obj interface{}) {
-	key, err := controller.KeyFunc(obj)
+	key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(obj)
 	if err != nil {
 		runtime.HandleError(fmt.Errorf("couldn't get key for object %#v: %v", obj, err))
 		return

